server/src/main/java/org/apache/druid/segment/indexing/ReaderUtils.java

```diff
@@ -60,6 +60,12 @@ public static Set<String> getColumnsRequiredForIngestion(
 
     // Find columns we need to read from the flattenSpec
     if (flattenSpec != null) {
+      if ((dimensionsSpec.getDimensionNames() == null || dimensionsSpec.getDimensionNames().isEmpty()) &&
```

Comment:
i think it would be better to check `dimensionsSpec.getDimensions` since `getDimensionNames` does a transform. Also, the list will never be null, so no need to check

Reply:
Done